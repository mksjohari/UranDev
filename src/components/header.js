import React from "react";
import { Link } from "gatsby";

import styles from "./modules/header.module.scss";
import Logo from "../../static/logo.png";

const Header = () => {
    return (
        <header className={styles.header}>
            <Link className={styles.title} to="/">
                <img
                    className={styles.imageLogo}
                    src={Logo}
                    alt="Uran Logo ..."
                />
                <span>URAN</span>
            </Link>
            <div className={styles.navContainer}>
                <nav>
                    <ul className={styles.navList}>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/"
                            >
                                Home
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/blog"
                            >
                                Blog
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/about"
                            >
                                About
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/contact"
                            >
                                Contact
                            </Link>
                        </li>
                    </ul>
                </nav>
            </div>
        </header>
    );
};

export default Header;
