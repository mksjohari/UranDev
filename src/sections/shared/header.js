import React from "react";
import { Link } from "gatsby";

import styles from "../../modules/header.module.scss";
import Logo from "../../../static/logo.png";

const Header = () => {
    return (
        <header className={styles.header}>
            <div className={styles.logoAndDetails}>
                <div className={styles.logo}>
                    <Link className={styles.headerLink} to="/">
                        <img
                            className={styles.imageLogo}
                            src={Logo}
                            alt="Uran Logo ..."
                        />
                        <h1 className={styles.textLogo}>URAN</h1>
                    </Link>
                </div>
                <div className={styles.details}>
                    <div className={styles.signUp}>Sign Up</div>
                    <div className={styles.login}>Login</div>
                </div>
            </div>
            <div className={styles.navContainer}>
                <nav>
                    <ul className={styles.navList}>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/"
                            >
                                Home
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/explore"
                            >
                                Explore
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/projects"
                            >
                                Projects
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/people"
                            >
                                People
                            </Link>
                        </li>
                    </ul>
                </nav>
            </div>
        </header>
    );
};

export default Header;
