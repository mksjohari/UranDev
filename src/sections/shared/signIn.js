import React, { useState } from "react";
import firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import { Link } from "gatsby";
import { Button } from "@material-ui/core";
import styles from "../../modules/signIn.module.scss";
import { getFirebase } from "../../firebase";
export const googleProvider = new firebase.auth.GoogleAuthProvider();

const signInWithGoogle = async () => {
    console.log("Sign in");
    getFirebase(firebase)
        .auth()
        .signInWithPopup(googleProvider)
        .then(function (result) {
            console.log(result);
        })
        .catch(function (error) {
            // Handle Errors here.
            const errorCode = error.code;
            const errorMessage = error.message;
            // The email of the user's account used.
            const email = error.email;
            // The firebase.auth.AuthCredential type that was used.
            const credential = error.credential;
            console.log(email, credential, errorCode, errorMessage);
            // ...
        });
};

const SignIn = () => {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [error] = useState(null);

    const signInWithEmailAndPasswordHandler = (event, email, password) => {
        event.preventDefault();
        console.log(event, email, password);
    };

    const onChangeHandler = (event) => {
        const { name, value } = event.currentTarget;

        if (name === "userEmail") {
            setEmail(value);
        } else if (name === "userPassword") {
            setPassword(value);
        }
    };
    console.log(getFirebase(firebase).auth().currentUser);
    return (
        <div className={styles.loginContainer}>
            <div className={styles.loginForm}>
                <h1>Sign In</h1>
                {error !== null && <div>{error}</div>}
                <form>
                    <label className={styles.label} htmlFor="userEmail">
                        Email:
                    </label>
                    <br />

                    <input
                        type="email"
                        name="userEmail"
                        value={email}
                        placeholder="E.g: uran@gmail.com"
                        id="userEmail"
                        onChange={(event) => onChangeHandler(event)}
                    />
                    <br />
                    <label className={styles.label} htmlFor="userPassword">
                        Password:
                    </label>
                    <br />

                    <input
                        type="password"
                        name="userPassword"
                        value={password}
                        placeholder="Your Password"
                        id="userPassword"
                        onChange={(event) => onChangeHandler(event)}
                    />
                    <br />
                    <Button
                        style={{ marginTop: 10 }}
                        color="primary"
                        variant="contained"
                        onClick={(event) => {
                            signInWithEmailAndPasswordHandler(
                                event,
                                email,
                                password
                            );
                        }}
                    >
                        Sign in
                    </Button>
                </form>
                <p className="text-center my-3">
                    Don't have an account? <Link to="/">Sign up here</Link>{" "}
                    <br /> <Link to="/">Forgot Password?</Link>
                </p>
                <hr />
                <Button
                    color="secondary"
                    variant="contained"
                    onClick={() => {
                        console.log("hello");
                        signInWithGoogle();
                    }}
                >
                    Sign in with Google
                </Button>
                {"   "}
                <Button
                    color="default"
                    variant="contained"
                    onClick={() => {
                        console.log(getFirebase(firebase).auth().currentUser);
                    }}
                >
                    Test Log
                </Button>
            </div>
        </div>
    );
};
export default SignIn;
