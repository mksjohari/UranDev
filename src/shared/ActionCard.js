import React from "react";
import { Draggable } from "react-beautiful-dnd";

import Options from "./sampleOptions";
import Dropdown from "./sandbox/Dropdown";
import styles from "../modules/tmp.module.scss";


class ActionCard extends React.Component {

  handleClick(event) {
    const task = document.getElementById(event.target.id.replace(/_editBtn/g,'') + '_edit');
    const type = event.target.className;

    if (type == "fas fa-edit") {
      task.style.display = "block";
      event.target.className = "fas fa-save";

      return;

    } else if (type == "fas fa-save") {
      task.style.display = "none";
      event.target.className = "fas fa-edit";

      return;
    }
  }


  render() {
    return (
      <Draggable 
        draggableId={this.props.action.id} 
        index={this.props.index}
      >
        {(provided) => (
          <div 
            {...provided.draggableProps}
            ref={provided.innerRef}
            className={styles.actionsCard}
          >
            <span className={styles.actionsContent}>
              <div className={styles.actions}>

              </div>

              <div className={styles.editAction} id={`${this.props.action.id}_edit`}>
                <div className={styles.tools} >
                  Used (tools)
                  <Dropdown 
                    options={Options} 
                    isMulti={true} 
                    isCreatable={true} 
                    keyColour="#AFDEE9"
                  />
                </div>
                <br />
                <div className={styles.tools} >
                  For (skills)
                  <Dropdown 
                    options={Options} 
                    isMulti={true} 
                    isCreatable={true}
                  />
                </div>
                <br />
                <div className={styles.actionDesc} >
                  Description (Optional)
                  <input className="inp-text"
                    placeHolder="type..."
                  ></input>
                </div>
              </div>
            </span>
            <div className={styles.btnCol}>
              <span {...provided.dragHandleProps} className={styles.dragHandle}>
                <i class="fas fa-grip-lines"></i>
              </span>
              <span className={styles.editBtn}>
                <i 
                  class="fas fa-edit" 
                  id={`${this.props.action.id}_editBtn`} 
                  onClick={this.handleClick}
                ></i>
              </span>
            </div>
          </div>
        )}
      </Draggable>
    );
  }
}

export default ActionCard;
