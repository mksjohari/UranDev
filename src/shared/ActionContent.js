import React, { Component } from "react";
import styles from "../modules/tmp.module.scss";

class ActionContent extends Component {


  render() {
    return (
      <>
        <div>{this.props.tools[0] ? 'Used tools' : ''}</div>
        <div className={this.props.tools[0] ? styles.tools : ''}>
          {this.props.tools[0] ?
            (this.props.tools.map((tool) => (
              <span className={styles.toolItem}>{tool}</span>
              )
            )) : <></>
          } 
        </div>
        <div>{this.props.skills[0] ? 'Used skills' : ''}</div>
        <div className={this.props.skills[0] ? styles.skills : ''}>
          {this.props.skills[0] ?
            (this.props.skills.map((skill) => (
              <span className={styles.skillItem}>{skill}</span>
              )
            )) : <></>
          } 
        </div>
        <div>{this.props.description.length ? 'Description:' : ''}</div>
        <div className={styles.description}>{this.props.description}</div>
      </>
    );
  }
}

export default ActionContent;
