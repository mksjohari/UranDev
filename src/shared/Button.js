import React from "react";

function Button(props) {
  return (
    <>
      <button
        type="button"
        className={props.colour}
        id={props.id}
        onClick={props.onClick}
      >
        {props.iconL ? props.iconL : ""}
        <span>{props.text}</span>
        {props.iconR ? props.iconR : ""}
      </button>
    </>
  );
}

export default Button;
