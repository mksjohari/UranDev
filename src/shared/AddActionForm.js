import React from 'react';

import Dropdown from './sandbox/Dropdown';
import Options from './sampleOptions';
import Button from './sandbox/Button';
import Droparea from './sandbox/Droparea';

import styles from '../modules/addActionForm.module.scss';

class AddActionForm extends React.Component {
  state = {};

  handleChange(event) {

    var newDesc = event.target.value.replace(/\n/g, "");
    event.target.value = newDesc;
    
    const newState = {
      ...this.state,
      action: {
        ...this.state.action,
        description: newDesc,
      }
    };

    this.setState(newState);
  }

  resetHeight(event) {
    event.target.style.height = 1 + "px";
    event.target.style.height = (5 + event.target.scrollHeight) + "px";
    // console.log(event.target.scrollHeight);
  }

  handleChange = this.handleChange.bind(this);

  render() {
    return (
      <>
        <div className={styles.container}>
          <h3>Add Action</h3>
          <div className={styles.toolsEdit} >
            <div className={styles.title} >What tool(s) did you use?</div>
            <Dropdown 
              id={this.props.actionid + '_tools'}
              options={Options.tools} 
              isMulti={true} 
              isCreatable={true} 
              colour="white"
              // defaultValue={this.state.action.tools.map(
              //   (tool) => (
              //     { value: tool, label: tool }
              //   )) }
              keyColour="#AFDEE9"
              onChange={this.handleSelectTool}
            />
          </div>
          <br />
          <div className={styles.skillsEdit} >
            <div className={styles.title} >What Skill(s) allowed you to achieve this?</div>
            <Dropdown 
              id={this.props.actionid + '_skills'}
              options={Options.skills} 
              isMulti={true} 
              isCreatable={true}
              colour="white"
              // defaultValue={this.state.action.skills.map(
              //   (skill) => (
              //     { value: skill, label: skill }
              //   )) }
              onChange={this.handleSelectSkill}
            />
          </div>
          <br />
          <div className={styles.actionDesc} >
            <div className={styles.title} >Describe your action in detail. (Optional)</div>
            <textarea 
              id={this.props.actionid + '_desc'}
              className={styles.descEdit}
              placeholder='Type...'
              onChange={this.handleChange}
              onKeyUp={this.resetHeight}
            ></textarea>
          </div>
          <br/>
          <div className={styles.actionFiles} >
            <div className={styles.title} >Add files to showcase your work in progress. (Optional, max 10 files)</div>
            <Droparea className={styles.actionFiles} />
          </div>

          <div className={styles.btnRow} >
            <Button
              colour='pink'
              id={this.props.id + '_add'}
              text="Confirm"
              iconR={<i className="fas fa-check" ></i>}
              onClick={this.props.addAction}
            />
            <Button
              colour='reddo'
              id={this.props.id + '_close'}
              text="Cancel"
              iconR={<i className="fas fa-times" ></i>}
              onClick={this.props.close}
            />
          </div>

        </div>
      </>
    )
  }
}

export default AddActionForm;
