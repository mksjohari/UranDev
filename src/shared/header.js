import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';
import styles from '../modules/header.module.scss';
import Logo from '../images/logo.png';
import Button from './sandbox/Button';
import { getFirebase } from './firebase/config';
import { logoutUser } from '../actions/userAction';
import Menu from './Menu';

function mapStateToProps(state) {
	return { user: state.user };
}

const Header = (props) => {
	const [user, setUser] = useState();
	useEffect(() => {
		if (props.user.uuid === '') {
			// check if logged, then call data
			getFirebase().auth().signOut();
			setUser(null);
		} else {
			setUser(props.user);
		}
	}, [props.user]);
	return (
		<header className={styles.header}>
			<div className={styles.logoAndDetails}>
				<div className={styles.logo}>
					<Link className={styles.headerLink} to="/">
						<img
							className={styles.imageLogo}
							src={Logo}
							alt="Uran Logo ..."
						/>
					</Link>
				</div>
				<div className={styles.details}>
					{user && <Menu />}
					{!user && (
						<div>
							<Button
								className={styles.signUp}
								text="Sign Up"
								onClick={props.toSignUp}
							/>
							<Button
								colour="yellow"
								text="Sign In"
								onClick={props.toLogin}
							/>
						</div>
					)}
				</div>
			</div>
			<div className={styles.navContainer}>
				<nav>
					<ul className={styles.navList}>
						<li>
							<NavLink
								activeClassName={styles.activeNavItem}
								className={styles.navItem}
								exact
								to="/"
							>
								Home
							</NavLink>
						</li>
						<li>
							<NavLink
								activeClassName={styles.activeNavItem}
								className={styles.navItem}
								exact
								to="/explore"
							>
								Explore
							</NavLink>
						</li>
						<li>
							<NavLink
								activeClassName={styles.activeNavItem}
								className={styles.navItem}
								to="/tmp"
							>
								HY's Wall of Shame
							</NavLink>
						</li>
					</ul>
				</nav>
			</div>
		</header>
	);
};

export default connect(mapStateToProps, { logoutUser })(Header);
