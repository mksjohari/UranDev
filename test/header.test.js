import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Header } from "../src/shared/header";
const renderer = require("react-test-renderer");
configure({ adapter: new Adapter() });

describe("<Header /> component runs without errors", () => {
    it("Header component renders without crashing", () => {
        shallow(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );
    });
    it("Header component (not logged in) matches previous snapshots", () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <Header user={{logged: false}} />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
