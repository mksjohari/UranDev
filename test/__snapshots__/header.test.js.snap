// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component runs without errors Header component (not logged in) matches previous snapshots 1`] = `
<header
  className="header"
>
  <div
    style={
      Object {
        "position": "absolute",
        "right": 800,
        "top": 100,
      }
    }
  >
    <button
      onClick={[Function]}
      type="button"
    >
      test Redux
    </button>
  </div>
  <div
    className="container"
  >
    <div
      className="headerContainer"
    >
      <a
        className="headerLink"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Uran Logo ..."
          className="imageLogo"
          src={
            Object {
              "process": [Function],
            }
          }
        />
        <div
          className="uran"
        >
          URAN
        </div>
      </a>
      <label
        id="label"
        onClick={[Function]}
      >
        <div
          id="nav_icon"
        >
          <span
            className=" nav_icon1 "
          />
          <span
            className=" nav_icon2 "
          />
          <span
            className=" nav_icon3 "
          />
          <span
            className=" nav_icon4 "
          />
        </div>
      </label>
    </div>
    <div
      className="navContainer"
    >
      <input
        className="toggle"
        id="toggle"
        type="checkbox"
      />
      <nav
        className="navBar"
      >
        <a
          aria-current="page"
          className="navItem activeNavItem"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="navItem"
          href="/explore"
          onClick={[Function]}
        >
          Explore
        </a>
        <div
          className="buttons"
        >
          <button
            className="undefined signUp"
            style={
              Object {
                "display": "inline",
              }
            }
            type="button"
          >
            
            <span>
              Sign Up
            </span>
            
            
          </button>
          <button
            className="yellow undefined"
            style={
              Object {
                "display": "inline",
              }
            }
            type="button"
          >
            
            <span>
              Sign In
            </span>
            
            
          </button>
        </div>
      </nav>
    </div>
  </div>
</header>
`;
